packages/api/src/services/emp-stats.ts

```diff
@@ -1,132 +1,155 @@
+    return Promise.allSettled(addresses.map(updateTvlHistory));
   }
 
   async function getPriceFromTable(empAddress: string, currencyAddress: string) {
     // PriceSample is type [ timestamp:number, price:string]
     const priceSample: PriceSample = prices[currency].latest[currencyAddress];
     assert(uma.utils.exists(priceSample), "No latest price found for emp: " + empAddress);
-    const [, price] = priceSample;
-    return price;
+    return priceSample;
   }
 
   async function getFullEmpState(empAddress: string) {
     const emp = await queries.getAnyEmp(empAddress);
     // the full state has collateral decimals, pulled from erc20 state
     return queries.getFullEmpState(emp);
   }
-  async function updateStats(empAddress: string) {
-    const update: empStats.Data = { address: empAddress };
-
-    const emp = await getFullEmpState(empAddress);
-    update.timestamp = nowS();
-
-    if (uma.utils.exists(emp.collateralCurrency)) {
-      const collateralPrice = await getPriceFromTable(empAddress, emp.collateralCurrency).catch(() => undefined);
-      if (collateralPrice !== undefined) {
-        update.collateralPrice = collateralPrice;
-        update.tvl = calcTvl(update.collateralPrice, emp).toString();
-      }
-    }
-
-    if (uma.utils.exists(emp.tokenCurrency)) {
-      const syntheticPrice = await getPriceFromTable(empAddress, emp.tokenCurrency).catch(() => undefined);
-      if (syntheticPrice !== undefined) {
-        update.syntheticPrice = syntheticPrice;
-        update.tvm = calcTvm(update.syntheticPrice, emp).toString();
-      }
-    }
-
-    const rawSyntheticPrice = await getSyntheticPriceFromTable(empAddress).catch(() => undefined);
-    if (rawSyntheticPrice !== undefined) {
-      // get the raw synth price for sanity check
-      update.rawSyntheticPrice = rawSyntheticPrice;
-    }
 
-    return stats[currency].latest.upsert(empAddress, update);
+  async function updateTvl(emp: uma.tables.emps.Data) {
+    assert(emp.collateralCurrency, "TVL Requires collateral currency for emp: " + emp.address);
+    const priceSample = await getPriceFromTable(emp.address, emp.collateralCurrency);
+    const value = await calcTvl(priceSample[1], emp).toString();
+    const update = {
+      value,
+      timestamp: priceSample[0],
+    };
+    return stats[currency].latest.tvl.upsert(emp.address, update);
+  }
+  async function updateTvm(emp: uma.tables.emps.Data) {
+    assert(emp.tokenCurrency, "TVL Requires token currency for emp: " + emp.address);
+    const priceSample = await getPriceFromTable(emp.address, emp.tokenCurrency);
+    const value = await calcTvm(priceSample[1], emp).toString();
+    const update = {
+      value,
+      timestamp: priceSample[0],
+    };
+    return stats[currency].latest.tvm.upsert(emp.address, update);
   }
-
   // update all stats based on array of emp addresses
-  async function updateAllStats(addresses: string[]) {
+  async function updateAllTvl(addresses: string[]) {
```

Comment:
minor side note: there seems to be a decent amount of repeated code in this class. Not necessarily a problem. Just wanted to note it.

Reply:
yea it eventually could be factored to share a single function under the hood, maybe. prefer to leave it due to time crunch